syntax = "proto3";

package mathtest;

// Defines the service
service MathTest {
    // Function invoked to send the request
    rpc sendRequest (MathRequest) returns (MathReply) {}
    rpc sendarray (MathRequest) returns (Mathhasil) {}
    rpc softhaliza (RecBro) returns (datablob) {}
}

// The request message containing requested numbers
message MathRequest {
    int32 a = 1;
    int32 b = 2;
}

// The response message containing response
message MathReply {
    int32 result = 1;
}

message Mathhasil {
    string header = 1;
    repeated float result_float = 2;
}

message RecBro {
    int32 MintaTipe = 1;
    int32 MintaParam = 2;
}

message datablob {
    int32 tipe_data = 1;
    int32 param = 2;
    int32 timeEpoch = 3;
    bytes dataku = 4;
}
